### PR TITLE
Fix build output for UD

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Fix build output for UD.

Please, try these cases:
1. mvn clean install -Pnative  (native, with tests)
2. mvn clean install   (with docker)

### CODE REVIEW

**Comment by olexii4** [comment]
ci-build

**Comment by olexii4** [comment]
ci-test

File: dashboard/Dockerfile [hunk 1]
```diff
@@ -20,9 +20,11 @@ RUN apt-get update && \
     && rm -rf /var/lib/apt/lists/*
 COPY package.json /dashboard/
 COPY yarn.lock /dashboard/
-RUN cd /dashboard && npm i yarn && npx yarn install --ignore-scripts
+COPY typings.json /dashboard/
+COPY package-lock.json /dashboard/
```
Comment (benoitf) [inline dashboard/Dockerfile]: hello, why do we have the need of package-lock.json as we're using yarn ?

**Review by ashumilova — APPROVED** [review]

**Comment by ashumilova** [comment]
should fix this one too https://github.com/eclipse/che/issues/11740

File: dashboard/Dockerfile (hunk 1 above)
Comment (ppitonak) [reply dashboard/Dockerfile]: @benoitf it's being discussed also in https://github.com/eclipse/che/pull/12923... the reason is that `npm i yarn` will generate this file. IMHO better approach is to get rid of `npm i yarn` so that the `package-lock.json` won't be generated at all

Comment (benoitf) [reply dashboard/Dockerfile]: @ppitonak yes but I prefer what is done in https://github.com/eclipse/che/pull/12923

Comment (ppitonak) [reply dashboard/Dockerfile]: I'm confused... do you mean the suggestion by @monaka ?

Comment (benoitf) [reply dashboard/Dockerfile]: not using package-lock and npm, but using only yarn

**Comment by ppitonak** [comment]
@olexii4 why did you duplicate basically all of my changes from https://github.com/eclipse/che/pull/12923?

File: dashboard/package.json [hunk 2]
```diff
@@ -84,14 +84,14 @@
     "webpack": "^1.12.11",
     "webpack-stream": "~2.1.0",
     "wrench": "~1.5.8",
-    "yarn": ">=1.2.1"
+    "yarn": "^1.13.0"
```
Comment (ppitonak) [inline dashboard/package.json]: how is this change related to "fix build output for UD"?

File: dashboard/Dockerfile (hunk 1 above)
Comment (olexii4) [reply dashboard/Dockerfile]: @benoitf  In this PR I lived an old solution, just fix needed yarn version in package-lock.json.  
Ok. I will change it(upgrade the yarn version in the image)

**Comment by olexii4** [comment]
@ppitonak  Yesterday I found that your changes are not complete. You didn't fix the native build "mvn clean install -Native". But I proved it two days before and didn't want to reject it. It is not critical( it broke the native build only) and I tried to fix it quietly: add these changes to my own PR. 
So I added the needed comment to your PR https://github.com/eclipse/che/pull/12923 recently.

**Comment by olexii4** [comment]
As I noted yesterday in https://github.com/eclipse/che/pull/12826

```
FYI: After reviewing your changes, I have created one more PR for live discussion all that was additionally done during checking. As for me, build output became much better after.
```

**Comment by olexii4** [comment]
I added some lines from bought  PRs(which was in progress) to avoid the merge conflicts.  I planned to merge after yours.  I solved another problem

File: dashboard/package.json (hunk 2 above)
Comment (olexii4) [reply dashboard/package.json]: I need to upgrade the yarn for using option.

Try to be shorter as possible:  you can try to build this branch with docker
Use this
```
RUN cd dashboard && yarn --ignore-optional
```
instead of
```
RUN cd dashboard && npm i yarn && ./node_modules/.bin/yarn yarn --ignore-optional
```

And you will have 'BUILD FAILURE' (option support with an old yarn version)

File: dashboard/Dockerfile (hunk 1 above)
Comment (olexii4) [reply dashboard/Dockerfile]: done

**Comment by olexii4** [comment]
ci-test

**Review by akurinnoy — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

File: dashboard/package.json (hunk 2 above)
Comment (olexii4) [reply dashboard/package.json]: I have removed "yarn" from the devDependencies at all.
